lightning/src/ln/channel.rs

```diff
@@ -4048,7 +4048,7 @@ impl<Signer: Sign> Channel<Signer> {
 		}
 
 		// Now update local state:
-		if (self.channel_state & (ChannelState::AwaitingRemoteRevoke as u32)) == (ChannelState::AwaitingRemoteRevoke as u32) {
+		if (self.channel_state & (ChannelState::AwaitingRemoteRevoke as u32 | ChannelState::MonitorUpdateFailed as u32)) != 0 {
```

Comment:
Can you update `send_htlc` doc in consequence, something-like "This returns an option instead of a pure UpdateAddHTLC as we may be in a state where we are waiting on the remote peer to send us a revoke_and_ack during which time we cannot add new HTLCs on the wire or a monitor is temporary under failure" ?